fsw/src/bpnode_app.h

```diff
@@ -0,0 +1,94 @@
+/************************************************************************
+ * NASA Docket No. GSC-18,719-1, and identified as “core Flight System: Bootes”
+ *
+ * Copyright (c) 2020 United States Government as represented by the
+ * Administrator of the National Aeronautics and Space Administration.
+ * All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ ************************************************************************/
+
+/**
+ * @file
+ *
+ * Main header file for the BPNode app
+ */
+
+#ifndef BPNODE_APP_H
+#define BPNODE_APP_H
+
+/*
+** Required header files.
+*/
+#include "cfe.h"
+#include "cfe_config.h"
+
+#include "bpnode_mission_cfg.h"
+#include "bpnode_platform_cfg.h"
+
+#include "bpnode_perfids.h"
+#include "bpnode_msgids.h"
+#include "bpnode_msg.h"
+
+/************************************************************************
+** Type Definitions
+*************************************************************************/
+
+/*
+** Global Data
+*/
+typedef struct
+{
+    /*
+    ** Command interface counters...
+    */
+    uint8 CmdCounter;
+    uint8 ErrCounter;
+
+    /*
+    ** Housekeeping telemetry packet...
+    */
+    BPNode_HkTlm_t HkTlm;
```

Comment:
Is there any reason not to use the command/error counters directly from the `HkTlm.Payload`? Why duplicate the data?

Currently it looks like we copy:

- `BPNode_AppData.CmdCounter` to `BPNode_AppData.HkTlm.Payload.CommandCounter`
- `BPNode_AppData.ErrCounter` to `BPNode_AppData.HkTlm.Payload.CommandErrorCounter`

Having two sets of counters (especially with different names) that track the same info seems confusing.

Reply:
Probably a relic from sampple_app. I can change that

Reply:
Done